File: FirstApp/app/src/main/java/com/example/firstapp/MainActivity.java
package com.example.firstapp;

import android.os.Bundle;
import android.view.View;
import android.widget.SeekBar;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import com.google.android.material.button.MaterialButton;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        TextView nameEdittext = findViewById(R.id.name_input);
        String name = nameEdittext.getText().toString();

        SeekBar seekBar = findViewById(R.id.seekbar);

        TextView info = findViewById(R.id.info);

        seekBar.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
            @Override
            public void onProgressChanged(SeekBar seekBar, int i, boolean b) {
                switch (i) {
                    case 0:
                        info.setText("malo");
                        break;
                    case 1:
                        info.setText("srednio");
                        break;
                    case 2:
                        info.setText("duzo");
                        break;
                }
            }

            @Override
            public void onStartTrackingTouch(SeekBar seekBar) {

            }

            @Override
            public void onStopTrackingTouch(SeekBar seekBar) {

            }
        });

        MaterialButton sendButton = findViewById(R.id.button);
        sendButton.setOnClickListener(view -> {
            if (name.length() < 5) {
                nameEdittext.setError("IMIE MUSI MIEC PRZYNAJMNIEJ 5 ZNAKOW!!!");
            }
        });
    }
}

File: X.12.25/ProbnyPikaczu/app/src/main/java/com/example/probnypikaczu/MainActivity.kt
package com.example.probnypikaczu

import android.os.Bundle
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.SeekBar
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.example.probnypikaczu.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()

        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        binding.seekBar.setOnSeekBarChangeListener(object : SeekBar.OnSeekBarChangeListener {
            override fun onProgressChanged(
                seekBar: SeekBar?,
                progress: Int,
                fromUser: Boolean
            ) {
                binding.imageCountText.text = getString(R.string.image_count, progress)
                binding.imageCountHeader.text = when (progress) {
                    1 -> {
                        getString(R.string.oneImage)
                    }

                    in 2..<5 -> {
                        getString(R.string.twotofourImages, progress)
                    }

                    else -> {
                        getString(R.string.overFiveImages, progress)
                    }
                }

                binding.imagesContainer.removeAllViews()

                for (i in 0 until progress) {
                    val imageView = ImageView(this@MainActivity)
                    imageView.setImageResource(R.drawable.pikaczu)

                    val params = LinearLayout.LayoutParams(
                        0,
                        LinearLayout.LayoutParams.WRAP_CONTENT,
                        1f
                    )
                    params.setMargins(8,8,8,8)
                    imageView.layoutParams = params
                    
                    binding.imagesContainer.addView(imageView)
                }
            }

            override fun onStartTrackingTouch(seekBar: SeekBar?) {

            }

            override fun onStopTrackingTouch(seekBar: SeekBar?) {

            }
        })
    }
}